hermes/bin/src/runtime_extensions/hermes/crypto/bip39.rs

```diff
@@ -0,0 +1,392 @@
+/// eg. 24 words will have 8 checksum bits.
+///
+/// # Arguments
+///
+/// - `word_count`: The number of words in the mnemonic. Must be a multiple of
+/// 3 and in the range of 12 - 24.
+/// - `prefix`: A vector of strings representing the prefix, empty if no prefix.
+/// - `language`: The language of the mnemonic.
+///
+/// # Returns
+///
+/// Returns the mnemonic of type `String` as a `Result`.
+/// If the conversion is successful, it returns `Ok` with the mnemonic.
+/// If there is an error during the computation, it returns `Err` with an `Errno`.
+///
+/// # Errors
+///
+/// - `InvalidMnemonicLength`: If the word count is not a multiple of 3 or not in the
+///   range of 12 - 24.
+/// - `PrefixTooLong`: If the prefix is longer than the maximum allowed length, max is 3.
+/// - `WordNotFound`: If a word in the mnemonic is not found in the word list.
+pub(crate) fn generate_new_mnemonic(
+    word_count: usize, prefix: Vec<String>, language: Option<String>,
+) -> Result<Vec<String>, Errno> {
+    // Check word count
+    if is_invalid_word_count(word_count) {
+        return Err(Errno::InvalidMnemonicLength);
+    }
+
+    // Number of prefix word should be <= 3.
+    if prefix.len() > 3 {
+        return Err(Errno::PrefixTooLong);
+    }
+
+    let language = string_to_language(&language.unwrap_or("English".to_string()));
+
+    let prefix_index_bits = match get_prefix_index_bits(prefix, language) {
+        Ok(prefix_index_bits) => prefix_index_bits,
+        Err(e) => return Err(e),
+    };
+
+    // Create an vec that will hold binary conversion of entropy.
+    let mut bits_entropy = Vec::new();
+    // Add the prefix index bit to the bit entropy.
+    bits_entropy.extend_from_slice(&prefix_index_bits);
+
+    let entropy = match generate_entropy(word_count) {
+        Ok(entropy) => entropy,
+        Err(e) => return Err(e),
+    };
+
+    // Convert bytes entropy to bits.
+    for byte in entropy {
+        for j in (0..8).rev() {
+            // Should not exceed the word_count / 3 * 4 * 8
+            // which is number of entropy bits for the mnemonic word count.
+            if bits_entropy.len() >= word_count / 3 * 4 * 8 {
+                break;
+            }
+            bits_entropy.push((byte >> j) & 1);
+        }
+    }
+
+    let check_sum_bits = get_check_sum_bits(&bits_entropy, word_count);
+    // Add the checksum bits to the end of bit entropy.
+    bits_entropy.extend_from_slice(&check_sum_bits);
+
+    let word_indices = get_word_indices(&bits_entropy, word_count);
+    let mnemonic_list = get_mnemonic_from_indices(word_indices, language);
+
+    Ok(mnemonic_list)
+}
+
+/// Check if the word count is valid.
+/// Valid word count is a multiple of 3 and in the range of 12 - 24.
+/// Returns true if the word count is invalid, otherwise false.
+fn is_invalid_word_count(word_count: usize) -> bool {
+    word_count < 12 || word_count % 3 != 0 || word_count > 24
+}
```

Comment:
I'd recommend not to negate a function name. It will be harder to interpret, for example, `!is_invalid_word_count` is simply `is_valid_word_count` but you have to interpret the code twice.

Reply:
I don't want to negate it when using the function might cause confusing so will stick to this, but changed the  function description to make it clearer

Reply:
I don't want to negate it when using the function might cause confusing so will stick to this, but changed the function description to make it clearer